apps/autorelayer-interop/src/relayer.ts

```diff
@@ -205,6 +155,258 @@ export class Relayer {
+
+    // Process all claims in parallel
+    await Promise.allSettled(
+      pendingClaims.map((claim) => this.processClaim(claim)),
+    )
+  }
+
+  /**
+   * Fetches pending claims for the given relayers
+   * @param relayers - The relayers to fetch pending claims for
+   * @returns The pending claims
+   * @throws Error if the pending claims fetch fails
+   */
+  private async fetchPendingClaims(
+    relayers: Address[],
+  ): Promise<PendingClaims> {
+    const relayersParam = JSON.stringify(relayers)
+    const url = `${
+      this.config.ponderInteropApi
+    }/messages/pending/claims?relayers=${encodeURIComponent(relayersParam)}`
+    const resp = await fetch(url, jsonFetchParams)
+    if (!resp.ok) {
+      throw new Error(`pending claims fetch http response: ${resp.statusText}`)
+    }
+
+    const body = await resp.json()
+    const { data: claims, error } = PendingClaimsSchema.safeParse(body)
+    if (error) {
+      throw new Error(
+        `api response: ${error.errors
+          .map((e) => `{${e.path.join('.')}: ${e.message}}`)
+          .join(', ')}`,
+      )
+    }
+    return claims
+  }
+
+  private async processClaim(claim: PendingClaim): Promise<void> {
+    const msgLog = this.log.child({
+      messageHash: claim.relayReceipt.messageHash,
+    })
+
+    const gasProvider = this.selectGasProvider(claim, msgLog)
+    if (!gasProvider) return
+
+    const clientsAndAccount = await this.getClientsAndAccount(
+      gasProvider.gasTankChainId,
+      msgLog,
+    )
+    if (!clientsAndAccount) return
+
+    const { client: publicClient, walletClient, account } = clientsAndAccount
+
+    const params = this.buildClaimParams(
+      claim,
+      account,
+      gasProvider.gasProviderAddress,
+    )
+
+    try {
+      const claimTxHash = await this.claimMessageViaGasTank({
+        publicClient,
+        walletClient,
+        gasTankAddress: this.config.gasTankAddress!,
+        params,
+        msgLog,
+      })
+      msgLog.info({ claimTxHash }, 'submitted message claim')
+    } catch (error) {
+      msgLog.warn('claim failed, skipping...')
+    }
+  }
+
+  private async getClientsAndAccount(
+    chainId: number,
+    msgLog: Logger,
+  ): Promise<{
+    client: PublicClient
+    walletClient: WalletClient
+    account: Account
+  } | null> {
+    if (!this.config.clients[chainId] || !this.config.walletClients[chainId]) {
+      msgLog.warn('no client for destination, skipping...')
+      return null
+    }
+
+    const client = this.config.clients[chainId]
+    const walletClient = this.config.walletClients[chainId]
+
+    // If no account is configured (sponsored), specify
+    // a random account owned by the sponsored endpoint.
+    let account = walletClient.account
+    if (!account) {
+      const addresses = await walletClient.getAddresses()
+      if (addresses.length === 0) {
+        msgLog.warn('no accounts found, skipping...')
+        return null
+      }
```

Comment:
❓ If I'm running a relayer, what's the use-case for using many addresses? Redundancy?

Reply:
I added a better explanation of this to the comment, but this is only needed for typescript purposes. In the sponsored case viem will send a `eth_sendTransaction` RPC to the sponsored sender service, and the sponsored sender service will handle signing and submitting the transaction